### DISCUSSION TITLE
Unable to pass in an AbortController signal to the HTTP Client

### DISCUSSION THREAD

**Discussion by donegjookim** [jsforce/jsforce#1469] [Ideas]
Correct me if I'm wrong, but there currently is no way to pass in an AbortController signal to the HTTP Client so that HTTP requests can be canceled. Can this be supported?

**Comment by cristiand391** [comment]
triaging old issues...

This is correct, jsforce doesn't take an AC for the HTTP client. 
It does use one internally to handle request timeouts but it's not exposed:
https://github.com/jsforce/jsforce/blob/b661afdd21729812974f1056e392f2f58fc21096/src/request.ts#L43

I would be interested in possible use-cases. We aren't going to expose this in jsforce v3 as we plan to remove `node-fetch` in the next major.

**Comment by dobesv** [comment]
I have been working on better graceful shutdown support in some of our tools and scripts, it would be helpful if we could supply an `AbortSignal` to jsforce operations so that if we are asked to shut down we can gracefully terminate the operation instead of relying on process exit to stop the operation.  It's not a major issue, just something that makes the output a bit cleaner.
